codebase/base-async/src/main/scala/com/virtuslab/base/async/IdentityHelpers.scala

```diff
@@ -29,14 +28,16 @@ trait IdentityHelpers extends DefaultJsonProtocol {
 
   private val identityUrl = s"http://${Config.identityServiceContactPoint}/api/v1/validate"
 
+  private val identityHttpClient = Http()
+
   def validateToken(token: String)(implicit traceId: TraceId): Future[Option[String]] = {
     val json = ValidateTokenRequest(token).toJson.compactPrint
     val entity = HttpEntity(`application/json`, json)
     val request = HttpRequest(POST, identityUrl)
       .withHeaders(RawHeader("X-Trace-Id", traceId.id))
       .withEntity(entity)
 
-    Http().singleRequest(request).flatMap { response =>
+    identityHttpClient.flatMapRequest(request) { response =>
       if (response.status.isSuccess())
         response.entity
           .dataBytes
```

Comment:
Isnt this going to fail? You are discarding entity bytes before this happens, aren't you?

Reply:
No, I don't think so, at least that was not my intent. 

```
    akkaHttp
      .singleRequest(httpRequest)
      .map { response =>
        val result = logic(response) // <--- here i take response and process it (syncronously)
        response.discardEntityBytes() // <--- here i discard
        result 
      }
```

Reply:
But then again - i might have just missed something obvious as i did all this change while participating in the call so i might have been sloppy somewhere (and I am still on the call :/ )

Reply:
We'll see if that helps then.